pkg/volume/emptydir/empty_dir_test.go

```diff
@@ -236,14 +243,19 @@ func doTestPlugin(t *testing.T, config pluginTestConfig) {
 		t.Errorf("Got a nil Unmounter")
 	}
 
-	// Tear down the volume
-	if err := unmounter.TearDown(); err != nil {
-		t.Errorf("Expected success, got: %v", err)
+	if config.idempotent {
```

Comment:
How come idempotent creates `volPath` but removes `metadataDir`?

Reply:
Also why is it removed before teardown?

Reply:
this idempotent setting is from previous PR, I am still not sure the exact reasoning behind it. It was used to indicate whether the Readyfile already exist or not before setting up the volume. In the previous design, readyFile could already exist because tear down does not delete it.

Now after changing the logic, I think maybe it is better to use different name since we are testing differently. Basically we have volumeDir, and readyDir, I am testing the following scenarios
1. neither volumeDir nor readyDir exist before volume setup
2. volumeDir exist, but not readyDir before setup
3. readyDir exist, volumeDir does not
4. both exist

For teardown, similarly it tests different scenarios to represent kubelet crashes or user intentionally delete dirs.
1. both dir exist
2. readyDir still exist, but not volumeDir
3. volumeDir still exist, but not readyDir

Reply:
updated the test, PTAL